### PR TITLE
Feature/connect kafka

### PR DESCRIPTION
**Pull request by annaoliinyk**
Changes mainly in kafka folder:
- docker-compose.yml -> instructions for kafka setup using docker;
- get_states_OpenSky.py -> with a function that returns a one random state in a form of dictionary with keys id (icao24) and values (all info about a state in form of a list);
- producer.py -> for writing states to a OpenSky_data_ingestion topic
- consumer.py -> for reading data from OpenSky_data_ingestion topic.

### CODE REVIEW

**Comment by annaoliinyk** [comment]
Ready for review

File: kafka_messenger/consumer.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class MyConsumer:
+    def __init__(self):
+        self.consumer = KafkaConsumer(
+            'OpenSky_data_ingestion',
+            bootstrap_servers='localhost:9092',
+            auto_offset_reset='earliest'
+        )
+
+    def run_consumer(self):
```
Comment (AnaKovaGH) [inline kafka_messenger/consumer.py:20]: For what purpose is this function?

File: kafka_messenger/consumer.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import sys
+
+from kafka import KafkaConsumer
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+class MyConsumer:
+    def __init__(self):
+        self.consumer = KafkaConsumer(
```
Comment (AnaKovaGH) [inline kafka_messenger/consumer.py]: The same code is in producer. It should be in parent class and inherited.

File: OpenSkyDataExtractor/get_states.py [hunk 3]
```diff
@@ -18,9 +18,21 @@ def __init__(self):
             logging.info("Exception occurred: ", e)
 
     def get_states(self):
-        credentials = get_credentials_from_file()
-        logging.info("Authenticating to https://opensky-network.org/api/states/all and getting all data for states")
-        states = requests.get("https://opensky-network.org/api/states/all", data=credentials).json()
+        try:
+            # try getting states as authenticated user:
+            credentials = get_credentials_from_file()
+            states = requests.get("https://opensky-network.org/api/states/all", data=credentials).json()
```
Comment (AnaKovaGH) [inline OpenSkyDataExtractor/get_states.py]: This link should be in configs file

File: OpenSkyDataExtractor/get_states.py [hunk 4]
```diff
@@ -18,9 +18,21 @@ def __init__(self):
...
+            logging.info("Logged in and authenticated to https://opensky-network.org/api/states/all and got all data "
+                         "for states")
+        except FileNotFoundError:
+            # try getting states as non-authenticated user:
+            states = requests.get("https://opensky-network.org/api/states/all").json()
```
Comment (AnaKovaGH) [inline OpenSkyDataExtractor/get_states.py]: Same here. Should be in configs file.

File: definitions.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import os
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
```
Comment (AnaKovaGH) [inline definitions.py]: It should be in the file where you need it

File: kafka_messenger/data_editor.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from random import randint
+
+
+class RandomStateExtractor:
```
Comment (AnaKovaGH) [inline kafka_messenger/data_editor.py]: Not random states, you need to upload all data in Kafka

File: kafka_messenger/get_states_OpenSky.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+from kafka_messenger.data_editor import RandomStateExtractor
+
+STATE_DICT_KEYS = ['icao24', 'values']
```
Comment (AnaKovaGH) [inline kafka_messenger/get_states_OpenSky.py]: Should be in config file

File: kafka_messenger/get_states_OpenSky.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class RandomState:
+    def get_random_state(self, data_ingestion_obj) -> dict:
+        # returns a dict with keys 'states' and 'time', many states and one time value:
+        states_dict = data_ingestion_obj.get_states()
```
Comment (AnaKovaGH) [inline kafka_messenger/get_states_OpenSky.py]: There is code duplication here

File: kafka_messenger/producer.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import json
+import logging
+import random
+import sys
+import time
+from datetime import datetime
+
+from kafka import KafkaProducer
+
```
Comment (AnaKovaGH) [inline kafka_messenger/producer.py:8]: There should be no space

File: kafka_messenger/producer.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+from kafka_messenger.get_states_OpenSky import RandomState
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+
+class MyProducer:
+    def __init__(self):
+        # Kafka Producer
+        self.producer = KafkaProducer(
```
Comment (AnaKovaGH) [inline kafka_messenger/producer.py:18]: Same as in producer

**Review by AnaKovaGH — CHANGES_REQUESTED** [review]
Please change what was commented

File: main.py [hunk 11]
```diff
@@ -1,19 +1,26 @@
 import logging
 import sys
 from datetime import datetime
-
+import kafka.errors
 from OpenSkyDataExtractor.get_states import DataIngestion
+from kafka_messenger import producer, consumer
```
Comment (AnaKovaGH) [inline main.py:6]: Move to kafka imports sections

File: main.py [hunk 12]
```diff
@@ -1,19 +1,26 @@
...
 logging.basicConfig(stream=sys.stdout, level=logging.INFO)
 
 
 def main():
     start_timestamp = datetime.now().timestamp()
     logging.info("Program started: {}".format(start_timestamp))
-    DataIngestion().print_states()
+    data_ingestion_obj = DataIngestion()
+    data_ingestion_obj.print_states()
     end_timestamp = datetime.now().timestamp()
```
Comment (AnaKovaGH) [inline main.py]: Move to the end of the program

File: main.py [hunk 13]
```diff
@@ -1,19 +1,26 @@
...
     logging.info("Program ended: {}".format(end_timestamp))
     logging.info("Program running took: {}".format(end_timestamp - start_timestamp))
+    try:
+        consumer.MyConsumer()
```
Comment (AnaKovaGH) [inline main.py]: How it should work?

File: main.py [hunk 14]
```diff
@@ -1,19 +1,26 @@
...
+        producer.MyProducer().send_all_states(data_ingestion_obj)
```
Comment (AnaKovaGH) [inline main.py]: Rewrite is without second data ingestion

**Review by AnaKovaGH — CHANGES_REQUESTED** [review]
Please review comments and rename all variables and class's names to be more readable

**Comment by annaoliinyk** [comment]
Ready for review

**Review by AnaKovaGH — APPROVED** [review]
LGTM!

**Pull request merged**